internal/cli/attach.go

```
@@ -14,63 +15,106 @@ import (
 	"github.com/mitchellh/cli"
 )
 
-// VersionCommand is the command to show the version of the agent
+// AttachCommand is the command to Connect to remote Bor IPC console
 type AttachCommand struct {
-	UI    cli.Ui
-	Meta  *Meta
-	Meta2 *Meta2
+	UI            cli.Ui
+	Meta          *Meta
+	Meta2         *Meta2
+	ExecCMD       string
+	Endpoint      string
+	PreloadJSFlag string
+	JSpathFlag    string
 }
 
 // Help implements the cli.Command interface
 func (c *AttachCommand) Help() string {
 	return `Usage: bor attach <IPC FILE>
 
-  Connect to Bor IPC console.`
+  Connect to remote Bor IPC console.`
 }
 
 // Synopsis implements the cli.Command interface
 func (c *AttachCommand) Synopsis() string {
 	return "Connect to Bor via IPC"
 }
 
+func (c *AttachCommand) Flags() *flagset.Flagset {
+
+	f := flagset.NewFlagSet("attach")
+
+	f.StringFlag(&flagset.StringFlag{
+		Name:  "exec",
+		Usage: "Command to run in remote console",
+		Value: &c.ExecCMD,
+	})
+
+	f.StringFlag(&flagset.StringFlag{
+		Name:  "preload",
+		Usage: "Comma separated list of JavaScript files to preload into the console",
+		Value: &c.PreloadJSFlag,
+	})
+
+	f.StringFlag(&flagset.StringFlag{
+		Name:  "jspath",
+		Usage: "JavaScript root path for `loadScript`",
+		Value: &c.JSpathFlag,
+	})
+
+	return f
+}
+
 // Run implements the cli.Command interface
 func (c *AttachCommand) Run(args []string) int {
 
-	c.remoteConsole(args)
+	flags := c.Flags()
+
+	//check if first arg is flag or IPC location
+	if len(args) == 0 {
+		args = append(args, "")
+	}
+	if args[0] != "" && string(args[0][0:2]) == "--" {
+		if err := flags.Parse(args); err != nil {
+			c.UI.Error(err.Error())
+			return 1
+		}
+	} else {
+		c.Endpoint = args[0]
+		if err := flags.Parse(args[1:]); err != nil {
+			c.UI.Error(err.Error())
+			return 1
+		}
+	}
+	c.remoteConsole()
```

Error is not being handled.